src/main/java/com/devcycle/sdk/server/local/managers/EnvironmentConfigManager.java

```diff
@@ -65,19 +68,57 @@ public boolean isConfigInitialized() {
     return config != null;
   }
 
-  private ProjectConfig getConfig() throws DVCException, JsonProcessingException {
+  private ProjectConfig getConfig() throws DVCException {
     Call<ProjectConfig> config = this.configApiClient.getConfig(this.sdkKey, this.configETag);
-
-    this.config = getConfigResponse(config);
+    this.config = getResponseWithRetries(config, 1);
     return this.config;
   }
 
-  private ProjectConfig getConfigResponse(Call<ProjectConfig> call) throws DVCException, JsonProcessingException {
+  private ProjectConfig getResponseWithRetries(Call<ProjectConfig> call, int maxRetries) throws DVCException {
+    // attempt 0 is the initial request, attempt > 0 are all retries
+    int attempt = 0;
+    do {
+      try {
+        return getConfigResponse(call);
+      } catch (DVCException e) {
+
+        attempt++;
+
+        // if out of retries or this is an unauthorized error, throw up exception
+        if ( !e.isRetryable() || attempt > maxRetries) {
+          throw e;
+        }
+
+        try {
+          // exponential backoff
+          long waitIntervalMS = (long) (10 * Math.pow(2, attempt));
+          Thread.sleep(waitIntervalMS);
+        } catch (InterruptedException ex) {
+          // no-op
+        }
+
+        // prep the call for a retry
+        call = call.clone();
+      }
+    }while (attempt <= maxRetries && pollingEnabled);
+
+    // getting here should not happen, but is technically possible
+    ErrorResponse errorResponse = ErrorResponse.builder().build();
+    errorResponse.setMessage("Out of retry attempts");
+    throw new DVCException(HttpResponseCode.SERVER_ERROR, errorResponse);
+  }
+
+  private ProjectConfig getConfigResponse(Call<ProjectConfig> call) throws DVCException {
     ErrorResponse errorResponse = ErrorResponse.builder().build();
     Response<ProjectConfig> response;
 
     try {
       response = call.execute();
+    } catch(JsonParseException badJsonExc) {
+      // Got a valid status code but the response body was not valid json,
+      // need to ignore this attempt and let the polling retry
+      errorResponse.setMessage(badJsonExc.getMessage());
+      throw new DVCException(HttpResponseCode.NO_CONTENT, errorResponse);
```

Comment:
I hate having to use exceptions for flow control but I am working around some limitations of the Retrofit2 library that is used to do the HTTP requests + json serialization. The test harness will return bad data with a 200 status code which is supposed to be ignorned, but the Retrofit library will explode on the bad data and not let us react to the status code first so this is a workaround